sdk/core/ts-http-runtime/src/policies/bearerTokenAuthenticationPolicy.ts

```diff
@@ -130,6 +136,47 @@ function getChallenge(response: PipelineResponse): string | undefined {
   return;
 }
 
+/**
+ * Checks if the request is allowed to be sent over an insecure connection.
+ *
+ * A request is allowed to be sent over an insecure connection when:
+ * - The `allowInsecureConnection` option is set to `true`.
+ * - The request has the `allowInsecureConnection` property set to `true`.
+ * - The request is being sent to `localhost` or `127.0.0.1`
+ */
+function allowInsecureConnection(
+  request: PipelineRequest,
+  options: BearerTokenAuthenticationPolicyOptions,
+): boolean {
+  if (options.allowInsecureConnection && request.allowInsecureConnection) {
+    const url = new URL(request.url);
+    if (url.hostname === "localhost" || url.hostname === "127.0.0.1") {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+/**
+ * Logs a warning about sending a bearer token over an insecure connection.
+ *
+ * This function will emit a node warning once, but log the warning every time.
+ */
+function emitInsecureConnectionWarning(): void {
+  const warning =
+    "Sending bearer token over insecure transport. Assume any token issued is compromised.";
+
+  coreLogger.warning(warning);
+
+  if (typeof process?.emitWarning === "function" && !emitInsecureConnectionWarning.warned) {
+    emitInsecureConnectionWarning.warned = true;
+    process.emitWarning(warning);
```

Comment:
Kinda nice I thought 

```
[vitest] (node:1358484) Warning: Sending bearer token over insecure transport. Assume any token issued is compromised.
[vitest] (Use `node --trace-warnings ...` to show where the warning was created)
```

Reply:
nice!  I'd rather spam everytime this is being used instead of just once though :D

Reply:
hehe yeah I was originally thinking the same but looks like as a best practice we're expected to only emit warning once https://nodejs.org/api/process.html#avoiding-duplicate-warnings

Figured if that was important enough to put in the docs it's probably the right approach? But I want to make some noise so I figure what we can land on is:
- Emit the warning once per node's docs
- log the warning every since time

So at least if you have logging configured to warn you can see it every time. Not sure if we prefer a differnet approach? I am happy to change it!